burn-core/src/optim/grads.rs

```diff
@@ -81,6 +85,11 @@ impl GradientsParams {
         module.visit(&mut visitor);
         grads_params
     }
+
+    /// Provide an iterator over the gradient tensors.
+    pub fn iter_mut(&mut self) -> impl Iterator<Item = (&ParamId, &mut TensorValue)> {
+        self.container.iter_mut()
+    }
```

Comment:
Is this still necessary? I would like to avoid exposing the `Box<dyn Any>` since we may change it later.

Reply:
Nope I'm getting rid of all that stuff in grad.rs and container.rs/